src/pages/_app.tsx

```diff
@@ -298,6 +275,59 @@ function MyApp({ Component, pageProps }: AppPropsWithLayout) {
 		}, [setBiconomyLoading, biconomyInitPromisesRef, initiateBiconomyUnsafe]
 	)
 
+	const importWebwallet = useCallback((privateKey: string) => {
+		const newWebwallet = new ethers.Wallet(privateKey);
+
+		// set the webwallet
+		localStorage.setItem('webwalletPrivateKey', privateKey);
+		setWebwallet(newWebwallet);
+
+		// reset everything else 
+		setLoadingNonce(false);
+		setNonce(undefined);
+		setScwAddress(undefined);
+		setBiconomyWalletClients({});
+		setBiconomyDaoObjs({});
+		setBiconomyLoading({});
+
+		biconomyInitPromisesRef.current = {};
+
+	}, [setWebwallet])
+
+	const exportWebwallet = useCallback(() => {
+
+		if(!webwallet) {
+			throw new Error('No webwallet to export')
+		}
+
+		return webwallet.privateKey
+
+	}, [webwallet])
+
+	useEffect(() => {
+		if(!webwallet) {
+			return
+		}
+
+		if(nonce && nonce !== 'Token expired') {
+			return
+		}
+
+		addAuthorizedUser(webwallet?.address)
```

Comment:
Probably best to use async/await syntax 

Also I think the code needs to be linted once?

Reply:
How to use async/await syntax inside useEffect?
I pushed a batch. Let me know if that's what you meant.

Reply:
async/await & promises are essentially the same thing, just differ in syntax. 

Eg of using this in a useEffect:
``` ts
useEffect(() => {
  (async() => {
    const abcd = await apiCall()
  })()
}, [])
```

Reply:
I don't understand the difference between what I'm currently doing and what you proposed. Please elaborate more.

Reply:
you're using `then()` and `catch()` instead of the async function

Reply:
I'm using an async function and applying `.then()` and `.catch()` (which I need to set a state). 

```
useEffect(() => {
  (async() => {
    const abcd = await apiCall()
  })().then(() => {})
  .catch(() => {})
}, [])
```

This is your code with `.then()` and `.catch()`.

Reply:
You don't need then and catch for this, keeping it simple:
``` ts
useEffect(() => {
  (async() => {
    try {
      await addAuthorizedUser(address)
      const newNonce = await getUseNonce()
      setNonce(newNonce)
    } catch(err) {
      logger.error({err}, 'some error')
    }
  })()
}, [])

```